tools/src/lib.rs

```diff
@@ -10,16 +10,32 @@ pub fn get_time() -> String {
 #[macro_export]
 macro_rules! tprint {
     ($($arg:tt)*) => {{
-        use tools::get_time;
-        println!("[{}] {}", get_time(), format!($($arg)*));
+        print!("[{}] ", $crate::get_time());
+        print!($($arg)*);
```

Comment:
Some debate could be done on the fact that calling twice some I/O function for one operation is pretty bad. :)